sky/cli.py

```diff
@@ -3737,6 +3737,17 @@ def jobs_launch(
 
     common_utils.check_cluster_name_is_valid(name)
 
+    reauth_needed_clouds = [
+        resource.cloud for task in dag.tasks for resource in task.resources if
+        resource.cloud is not None and resource.cloud.can_credential_expire()
+    ]
+    if len(reauth_needed_clouds) > 0:
+        prompt = (
+            f'Launching jobs with cloud(s) {reauth_needed_clouds} may lead to '
+            'jobs being out of control. It is recommended to use credentials  '
+            'that never expire or a service account. Proceed?')
+        click.confirm(prompt, default=False, abort=True, show_default=True)
```

Comment:
Can we make this a warning instead of a confirmation prompt?

Reply:
use a yellow warning instead